### PR TITLE
Moved buttons into the registry.templates under ButtonTemplates

### PR DESCRIPTION
**Pull request by heath-freenome**
### Reasons for making this change

Inspired by an OLD PR https://github.com/rjsf-team/react-jsonschema-form/pull/1074 fixing https://github.com/rjsf-team/react-jsonschema-form/issues/567 but made way better

- Updated `@rjsf/utils` `types.ts` as follows:
  - Added two new types for buttons `SubmitButtonProps` and `IconButtonProps`
  - Updated `TemplatesType` to add a new `ButtonTemplates` sub-section that contains the `AddButton`, `MoveDownButton`, `MoveUpButton`, `RemoveButton` and `SubmitButton`
  - Updated `getTemplate()` to only return `ButtonTemplates` from the `registry`
- Update `@rjsf/core` to move the buttons into the `templates` as follows:
  - Moved `AddButton` into the `src/templates/ButtonTemplates` directory, fixing up the type to use as `IconButtonProps`
  - Moved `SubmitButton` into the `src/templates/ButtonTemplates` directory, fixing up the type to use as `SubmitButtonProps`
  - Moved `IconButton.js` into the `src/templates/ButtonTemplates` directory, converting it to Typescript
    - Also exported the `MoveDownButton`, `MoveUpButton` and `RemoveButton` components as the specific implementations of `IconButton` derived from the other templates
  - Updated `templates/index.ts` to create the `ButtonTemplates` inner object for `templates`
  - Updated `widgets/index.ts` to remove `SubmitButton` from it
  - Updated `WrapIfAdditional` to pull the `RemoveButton` from the `registry.templates.ButtonTemplates` and use it in place of the `IconButton`
  - Updated `ArrayFieldItemTemplate` to refactor the `MoveDownButton`, `MoveUpButton` and `RemoveButton`s into `IconButton`, pulling them out of `registry.templates.ButtonTemplates`
  - Updated `ArrayFieldTemplate` to use `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `ObjectFieldTemplate` to use `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `Form` to pull `SubmitButton` from `registry.templates.ButtonTemplates`
  - Updated the main `index.ts` file to no longer export `AddButtonProps`
- Updated `@rjsf/antd` to move the buttons into the `templates` as follows:
  - Renamed the `SubmitButton.js` file to `index.js` to match the naming convention in the package for files, while moving it from `widget` to `templates`
  - Created the `IconButton/index.js` file in `src/templates` by refactoring the `AddButton`, `MoveDownButton`, `MoveUpButton` and `RemoveButton` implementations from other templates
  - Updated `ArrayFieldItemTemplate` to use the `MoveDownButton`, `MoveUpButton` and `RemoveButton` from `registry.templates.ButtonTemplates`
    - Added `title` properties on all icons, removing `Add Item` text shown to the user
  - Updated `ArrayFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `ObjectFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `FieldTemplate` to pass `registry` to `WrapIfAdditional`
  - Updated `WrapIfAdditional` to use the `RemoveButton` from `registry.templates.ButtonTemplates`
  - Updated the main `index.js` to properly export the `templates.ButtonTemplates`
  - Updated the test snapshots due to the `IconButton` refactor
- Updated `@rjsf/bootstrap-4` to move the buttons into the `templates` as follows:
  - Updated `AddButton` to use `IconButtonProps` instead of `AddButtonProps`, adding a `title` to match core
  - Updated `SubmitButton` to use the `SubmitButtonProps` type
  - Updated `IconButton` to export `MoveDownButton`, `MoveUpButton` and `RemoveButton` components as the specific implementations of `IconButton` derived from the other templates
  - Updated `Templates.ts` to create the `ButtonTemplates` inner object for `templates`
  - Updated `Widgets.ts` to remove `SubmitButton` from it
  - Updated `ArrayFieldItemTemplate` to refactor the `MoveDownButton`, `MoveUpButton` and `RemoveButton`s into `IconButton`, pulling them out of `registry.templates.ButtonTemplates`
  - Updated `ArrayFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `ObjectFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `FieldTemplate` to pass `registry` to `WrapIfAdditional`
  - Updated `WrapIfAdditional` to use the `RemoveButton` from `registry.templates.ButtonTemplates`
  - Updated test snapshots to pick up the little differences due to the `IconButton` refactor
    - Also deleted the `IconButton.test.ts` because it wasn't needed
- Updated `@rjsf/chakra-ui` to move the buttons into the `templates` as follows:
  - Updated `AddButton` to use `IconButtonProps` instead of `AddButtonProps`, moving the text to a `title` to match core
  - Updated `SubmitButton` to use the `SubmitButtonProps` type
  - Updated `IconButton` to export `MoveDownButton`, `MoveUpButton` and `RemoveButton` components as the specific implementations of `IconButton` derived from the other templates
  - Updated `Templates.ts` to create the `ButtonTemplates` inner object for `templates`
  - Updated `Widgets.ts` to remove `SubmitButton` from it
  - Updated `ArrayFieldItemTemplate` to refactor the `MoveDownButton`, `MoveUpButton` and `RemoveButton`s into `IconButton`, pulling them out of `registry.templates.ButtonTemplates`
  - Updated `ArrayFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `ObjectFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `FieldTemplate` to pass `registry` to `WrapIfAdditional`
  - Updated `WrapIfAdditional` to use the `RemoveButton` from `registry.templates.ButtonTemplates`
  - Updated test snapshots to pick up the little differences due to the `IconButton` refactor
- Updated `@rjsf/fluent-ui` to move the buttons into the `templates` as follows:
  - Updated `AddButton` to use `IconButtonProps` instead of `AddButtonProps`, moving the text to a `title` to match core
  - Updated `SubmitButton` to use the `SubmitButtonProps` type
  - Updated `IconButton` to export `MoveDownButton`, `MoveUpButton` and `RemoveButton` components as the specific implementations of `IconButton` derived from the other templates
  - Updated `Templates.ts` to create the `ButtonTemplates` inner object for `templates`
  - Updated `Widgets.ts` to remove `SubmitButton` from it
  - Updated `ArrayFieldItemTemplate` to refactor the `MoveDownButton`, `MoveUpButton` and `RemoveButton`s into `IconButton`, pulling them out of `registry.templates.ButtonTemplates`
  - Updated `ArrayFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated test snapshots to pick up the little differences due to the `IconButton` refactor
- Updated `@rjsf/material-ui` and `@rjsf/mui` to move the buttons into the `templates` as follows:
  - Updated `AddButton` to use `IconButtonProps` instead of `AddButtonProps`, moving the text to a `title` to match core
  - Updated `SubmitButton` to use the `SubmitButtonProps` type
  - Implemented `IconButton` to export `MoveDownButton`, `MoveUpButton` and `RemoveButton` components as the specific implementations of `IconButton` derived from the other templates
  - Updated `Templates.ts` to create the `ButtonTemplates` inner object for `templates`
  - Updated `Widgets.ts` to remove `SubmitButton` from it
  - Updated `ArrayFieldItemTemplate` to refactor the `MoveDownButton`, `MoveUpButton` and `RemoveButton`s into `IconButton`, pulling them out of `registry.templates.ButtonTemplates`
  - Updated `ArrayFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `ObjectFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `FieldTemplate` to pass `registry` to `WrapIfAdditional`
  - Updated `WrapIfAdditional` to use the `RemoveButton` from `registry.templates.ButtonTemplates`
  - Updated test snapshots to pick up the little differences due to the `IconButton` refactor
- Updated `@rjsf/semantic-ui` to move the buttons into the `templates` as follows:
  - Updated `AddButton` moving the text to a `title` to match core
  - Updated `IconButton` to export `MoveDownButton`, `MoveUpButton` and `RemoveButton` components as the specific implementations of `IconButton` derived from the other templates
  - Updated `Templates.ts` to create the `ButtonTemplates` inner object for `templates`
  - Updated `Widgets.ts` to remove `SubmitButton` from it
  - Updated `ArrayFieldItemTemplate` to refactor the `MoveDownButton`, `MoveUpButton` and `RemoveButton`s into `IconButton`, pulling them out of `registry.templates.ButtonTemplates`
  - Updated `ArrayFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `ObjectFieldTemplate` to use the `AddButton` from `registry.templates.ButtonTemplates`
  - Updated `FieldTemplate` to pass `registry` to `WrapIfAdditional`
  - Updated `WrapIfAdditional` to use the `RemoveButton` from `registry.templates.ButtonTemplates`
  - Updated test snapshots to pick up the little differences due to the `IconButton` refactor

NOTE: This is stacked on top of #3007 and should be rebased after that PR merged

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/core/src/components/templates/FieldTemplate/WrapIfAdditional.tsx [hunk 1]
```diff
@@ -64,11 +66,8 @@ export default function WrapIfAdditional<T = any, F = any>(
         </div>
         <div className="form-additional form-group col-xs-5">{children}</div>
         <div className="col-xs-2">
-          <IconButton
-            type="danger"
-            icon="remove"
+          <RemoveButton
             className="array-item-remove btn-block"
-            tabIndex="-1"
```
Comment (nickgros) [inline packages/core/src/components/templates/FieldTemplate/WrapIfAdditional.tsx:71]: Wonder why tab navigation was disabled here...

File: packages/semantic-ui/src/IconButton/IconButton.js [hunk 2]
```diff
@@ -3,8 +3,16 @@ import PropTypes from "prop-types";
...
 
 function IconButton(props) {
-  const { icon, className, ...otherProps } = props;
-  return <Button icon={icon} className={className} {...otherProps} />;
+  const { icon, iconType, className, ...otherProps } = props;
+  return (
+    <Button
+      icon={icon}
+      size={iconType}
+      tabIndex="-1"
```
Comment (nickgros) [inline packages/semantic-ui/src/IconButton/IconButton.js:11]: Let's remove the tabIndex="-1" added here--all buttons in every theme should be accessible

**Review by nickgros — APPROVED** [review]
See comments for followup. Requested change is about behavior that did not change in this PR, so I'm approving.

File: packages/semantic-ui/src/IconButton/IconButton.js (hunk 2 above)
Comment (heath-freenome) [reply packages/semantic-ui/src/IconButton/IconButton.js]: Done!

**Pull request merged**